R/utils.R

```diff
@@ -485,15 +472,122 @@ multi_choice_to_labels <- function(db_data, db_metadata) {
 
       # Replace values from db_data$(field_name) with label values from
       # parse_labels key
-      db_data[[field_name]] <- db_data[[field_name]] %>%
-        as.character() %>%
-        recode(!!!parse_labels_output) %>%
-        factor(levels = unique(parse_labels_output))
+
+      db_data[[field_name]] <- label_handler(
+        x = db_data[[field_name]],
+        labels = parse_labels_output,
+        ptype = db_data[[field_name]]
+      )
     }
   }
   db_data
 }
 
+#' @title
+#' Apply factor labels to a vector
+#'
+#' @param x a vector to label
+#' @param labels a named vector of labels in the format `c(value = label)`
+#' @param \dots unused
```

Comment:
Do we even need `...` as an option then? Is this just for future-proofing?

Reply:
In this case we do need it because this function gets called as `label_handler` which takes a `ptype` when `label_handler <- apply_labs_haven`. The `...` just allows that `ptype` to get passed through unused. I'll make the annotation clearer.